phones/models.py

```diff
@@ -155,13 +155,16 @@ class RelayNumber(models.Model):
     vcard_lookup_key = models.CharField(
         max_length=6, default=vcard_lookup_key_default, unique=True
     )
+    enabled = models.BooleanField(default=True)
```

Comment:
All new fields either need `nullable=True`, or we need to set a server default in the migration. The `nullable=True` has been a real pain, let's try to figure out the server default...

Reply:
Hrm ... when I ran the migration on my local it didn't complain. Maybe I didn't have any existing records when I ran it? Or does django know enough that if `default=True` is in the field definition, it sets all existing records to `True` ?

Reply:
The problem is not running the migration, the problem is when we deploy it. The Kubernetes rollout preserves the number of web pods by replacing old ones with new ones, one at a time, so the old pods will run against the new database schema. You can replicate this with:

1. Check out this PR
2. Run migration 0017
3. Check out tag `v3.6.1` (or `main`, close enough)
4. Go through the steps to create a Relay number (register a real phone, setup a relay number)

Steps 4 is hard, but you'll end up where the old code is running:

```sql
INSERT INTO "phones_relaynumber"
   ("id", "number", "location", "user_id", "vcard_lookup_key")
   VALUES (123, '+1234567890', 'location', 456, 'acb123');
```

resulting in the error:

```
django.db.utils.IntegrityError: phones_relaynumber.enabled NOT NULL constraint failed
```

It was a little easier to see with PR #2232, where a integer column was added, and you could use the UI to fail to create a new alias.

So, we'll get database errors and data loss during the rollout. There's two ways to deal with this. One is to add `null=True`, the other is a database default.

The deploy sequence with `null=True`:

1. Deploy One:
    - A migration adds a column with `NULL` allowed.
    - Old code omits the new column on inserts and updates, and new rows get a `null` value.
    - The new code needs to expect a `null` value and interpret it as the default (in this case `True`).
2. Deploy Two:
    - A data migration replaces all `NULL` values with `True`, and a schema migration changes `NULL` to `NOT NULL`. 
    - The old code continues to set to `true` or `false`, and checks for `null` even though is won't happen.
    -  The new code can drop the `null` checks and assume only `true` or `false`.

The deploy sequence with a database default:

1. Deploy One:
   - A migration adds a column with a database default
   - Old code omits the new column on inserts and updates, and new rows get the database default
   - The new code gets either `true` or `false`, never `null`
2. Deploy Two:
   - A migration drops the database default
   - The old code and the new code are identical, always setting the column to `true` or `false`
   
The database default is a lot cleaner - the code doesn't change, and no mucking about with `null` values. The problem is that Django doesn't support database defaults ([issue #470](https://code.djangoproject.com/ticket/470)), so you can't do it without writing your own migrations. And _that_ is hard because we have to support both SQLite3 and PostgreSQL.

You can see what Django does with `./manage.py sqlmigrate phone 0017`:

```
BEGIN;
--
-- Add field enabled to relaynumber
--
CREATE TABLE "new__phones_relaynumber" ("id" integer NOT NULL PRIMARY KEY AUTOINCREMENT, "enabled" bool NOT NULL, "number" varchar(15) NOT NULL, "location" varchar(255) NOT NULL, "user_id" integer NOT NULL REFERENCES "auth_user" ("id") DEFERRABLE INITIALLY DEFERRED, "vcard_lookup_key" varchar(6) NOT NULL UNIQUE);
INSERT INTO "new__phones_relaynumber" ("id", "number", "location", "user_id", "vcard_lookup_key", "enabled") SELECT "id", "number", "location", "user_id", "vcard_lookup_key", 1 FROM "phones_relaynumber";
DROP TABLE "phones_relaynumber";
ALTER TABLE "new__phones_relaynumber" RENAME TO "phones_relaynumber";
CREATE INDEX "phones_relaynumber_number_742e5d6b" ON "phones_relaynumber" ("number");
CREATE INDEX "phones_relaynumber_user_id_62c65ede" ON "phones_relaynumber" ("user_id");
COMMIT;
```

The sequence is:
* Create a new table `new__phones_relaynumber` with the `enable` column
* Copy all the data from `phones_relaynumber`, setting the `enable` column to `1` (true)
* Drop `phones_relaynumber`
* Rename `new__phones_relaynumber` to `phones_relaynumber`
* Re-add the indexes

This is what the Django docs say for [sqlite3 migrations](https://docs.djangoproject.com/en/3.2/topics/migrations/#sqlite):

> SQLite has very little built-in schema alteration support, and so Django attempts to emulate it by:
>
> * Creating a new table with the new schema
> * Copying the data across
> * Dropping the old table
> * Renaming the new table to match the original name

I _think_ the SQL for PostgreSQL looks like:

```
ALTER TABLE "phones_relaynumber" ADD COLUMN "enabled" NOT NULL DEFAULT "true";
ALTER TABLE "phones_relaynumber" ALTER COLUMN "enabled" DROP DEFAULT;
```

but it is hard to get PostgreSQL running locally. I'll need to do that for this PR.

There's a note for [PostgreSQL](https://docs.djangoproject.com/en/3.2/topics/migrations/#postgresql):

> The only caveat is that prior to PostgreSQL 11, adding columns with default values causes a full rewrite of the table, for a time proportional to its size. For this reason, it’s recommended you always create new columns with null=True, as this way they will be added immediately.

We're running PostgreSQL 11.15.